dspy/primitives/prompt.py

```diff
@@ -0,0 +1,22 @@
+from typing import Any, Optional
+
+from pydantic import BaseModel
+
+
+class Prompt(BaseModel):
+    content: Optional[str]
+    messages: Optional[list[dict[str, Any]]]
+
+    @property
+    def is_valid(self):
+        return self.content is not None or self.messages is not None
+
+    def get_messages(self):
+
+        if not self.is_valid:
+            raise ValueError("Prompt has not been initialized with a content or messages")
+
+        if self.messages is not None:
+            return self.messages
```

Comment:
Would it be simpler to have `Prompt` only have `messages` and then make a subclass `SimplePrompt` which just has `content` and wraps it as `[{"role": "user", "content": self.content}]`?

Reply:
As some models presumably would take in just a string, I imagine we want a structure in which we can provide either a text prompt, or ChatML messages. I was thinking of managing this with simpler constructors, ie. something like this:

`Prompt.from_str(content: str)`

And then having the LM identify what it needs from the Prompt.
Im pretty easy on whatever this looks like, just want to prefer keeping less classes, over having one class for each variant.